### PR TITLE
[DS-1653] - Initial implementation of the WS Override plugin Manager

### PR DESCRIPTION
**Pull request by aleevas**
Jira issue [DS-1653](https://yusa.atlassian.net/browse/DS-1653)

How to check:

- [ ] In the custom module please create a new yml file with `{module_name}.ws_lb_override.yml` the next structure:
```yml
{module_name}.{component_id}_override:
  block_type: lb_testimonials
  library: {module_name}/lb_testimonials_override
  override: false
  active_colorways:
    - small_y*
    - '*green'
```

- [ ] clear the cache
- [ ] create a new library for override rules
image

- [ ] In the files for this library add a rule to override (just for example)
image

- [ ] go to the LP with LB
- [ ] in the LB settings, please select WS Samll colorway
- [ ] save the LB
- [ ]  verify that your override rules are applied

### CODE REVIEW

**Review by rollki — APPROVED** [review]
LGTM, thanks @aleevas

File: src/WSOverrideLayoutBuilder.php [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function isApplicableForColorway(EntityInterface $entity): bool {
+    foreach ($this->getDefinitions() as $definition) {
+      if (!empty($definition['active_colorways'])) {
+        $selected_colorway = $this->getSelectedColorway($entity);
+        $active_colorways = $definition['active_colorways'];
+        foreach ($active_colorways as $active_colorway) {
+          if (str_contains($active_colorway, '*')) {
```
[the remark this answers is not in the document]
Comment (aleevas) [reply src/WSOverrideLayoutBuilder.php]: @anpolimus improved at this commit [af14e53be0c5f1b2966b2a48c014e800ec8b46aa](https://github.com/YCloudYUSA/y_lb/pull/245/commits/af14e53be0c5f1b2966b2a48c014e800ec8b46aa)
cc @rollki

**Review by podarok — CHANGES_REQUESTED** [review]
this will be 3.8.0 version of y_lb and https://git.drupalcode.org/project/ws_small_y/-/merge_requests/1 should have dependency in composer.json from y_lb >= 3.8

**Review by anpolimus — APPROVED** [review]

File: src/WSOverrideLayoutBuilder.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (anpolimus) [reply src/WSOverrideLayoutBuilder.php]: @aleevas thank you. Now this PR is approved from my end.

**Review by rollki — APPROVED** [review]

**Review by podarok — APPROVED** [review]

**Pull request merged**